RecoTracker/Configuration/python/customiseEarlyDeleteForCKF.py

```diff
@@ -0,0 +1,24 @@
+import FWCore.ParameterSet.Config as cms
+
+import collections
+
+def customiseEarlyDeleteForCKF(process, products):
+
+    if "trackExtenderWithMTD" not in process.producerNames():
+        return products
+
+    def _branchName(productType, moduleLabel, instanceLabel=""):
+        return "%s_%s_%s_%s" % (productType, moduleLabel, instanceLabel, process.name_())
+
+    for name, module in process.producers_().items():
+        cppType = module._TypedParameterizable__type
+        if cppType == "TrackProducer":
+            if module.TrajectoryInEvent:
+                products[name].append(_branchName("Trajectorys", name))
+                products[name].append(_branchName("TrajectorysToOnerecoTracksAssociation", name))
```

Comment:
Just to explain a bit more, these lines "declare" data products produced by this module that will be deleted early. The machinery in `earlyDeleteSettings_cff.py` then adds the necessary configuration parameters (to tell the early deletion system that the product needs to be kept alive until that module has been run) to all modules whose configuration contain `cms.InputTag` referring to it.

(and thus any module that has a hardcoded `InputTag`, or `InputTag` parameter getting added in the [configuration validation step](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp) would break it)

Reply:
> So the only issue is in case one of these products should be made persistent, am I correct? Is there any built-in cross check to prevent conflicts in this case?

By quick read on the code data products consumed by output modules are not deleted early.

Reply:
ok, according to your last sentence persistency seems not the only possible issue...